### PR TITLE
add aqua test

### PR DESCRIPTION
**Pull request by Roger-luo**
It seems we have quite a lot ambiguity errors

### CODE REVIEW

**Comment by Roger-luo** [comment]
all ambiguity fixed, gonna merge after CI pass

File: src/bit_str.jl [hunk 1]
```diff
@@ -31,23 +34,35 @@ julia> bit"1101"[2]
...
 end
 
 const BitStr64{N} = BitStr{N,Int64}
 const LongBitStr{N} = BitStr{N,BigInt}
 
-BitStr{N,T}(val::BitStr{N,T}) where {N,T} = val
+BitStr{N,T}(val::BitStr) where {N,T<:Integer} = convert(BitStr{N,T}, val)
+BitStr{N,T}(val::BitStr{N,T}) where {N,T<:Integer} = val
+
+# for Type in [Complex, Rational, Enum, AbstractChar, BigFloat, Base.TwicePrecision]
```
Comment (GiggleLiu) [inline src/bit_str.jl:50]: Maybe you forgot to remove these lines?

File: src/bit_str.jl [hunk 2]
```diff
@@ -31,23 +34,35 @@ julia> bit"1101"[2]
...
+#     @eval (::Type{T})(val::$Type) where {T <: BitStr} = error("cannot convert $(typeof(val)) to BitStr")
+# end
+
 Base.zero(::Type{BitStr{N,T}}) where {N,T} = BitStr{N,T}(zero(T))
 Base.zero(::BitStr{N,T}) where {N,T} = BitStr{N,T}(zero(T))
 
 buffer(b::BitStr) = b.buf
 Base.reinterpret(::Type{BitStr{N,T}}, x::Integer) where {N,T} = BitStr{N,T}(reinterpret(T, x))
 Base.reinterpret(::Type{T}, x::BitStr) where {T} = reinterpret(T, buffer(x))
+Base.reinterpret(::Type{BitStr{N,T}}, x::BitStr) where {N,T} = x
```
Comment (GiggleLiu) [inline src/bit_str.jl:60]: The output type is incorrect.

File: src/bit_str.jl [hunk 3]
```diff
@@ -83,11 +98,13 @@ for op in [:<, :>, :(<=), :(>=)]
     @eval Base.$op(a::T, b::T) where {T<:BitStr} = Base.$op(buffer(a), buffer(b))
 end
 
-for op in [:(==)]
-    @eval Base.$op(a::T, b::Number) where {T<:BitStr} = Base.$op(buffer(a), b)
-    @eval Base.$op(a::Number, b::T) where {T<:BitStr} = Base.$op(a, buffer(b))
-    @eval Base.$op(a::BitStr{N}, b::BitStr{N}) where {N} = Base.$op(buffer(a), buffer(b))
+for Type in [Number, BigInt, BigFloat, AbstractIrrational, Rational, Complex]
+    @eval Base.:(==)(a::T, b::$Type) where {T<:BitStr} = Base.:(==)(buffer(a), b)
+    @eval Base.:(==)(a::$Type, b::T) where {T<:BitStr} = Base.:(==)(a, buffer(b))
```
Comment (GiggleLiu) [inline src/bit_str.jl:103]: Maybe we only want to compare `BitStr` and `Integer` types. e.g. It does not make sense to compare BitStr and complex numbers.

File: src/reorder.jl [hunk 4]
```diff
@@ -11,15 +11,34 @@ struct ReorderedBasis{N,T<:Integer}
...
 
+function ReorderedBasis(
+    orders::Tuple{T, Vararg{T, N}},
+    takers::Tuple{T, Vararg{T, N}},
+    differ::Tuple{T, Vararg{T, N}},
+) where {N,T}
+    ReorderedBasis{N+1,T}(orders, takers, differ)
+end
+
+# ReorderedBasis(orders::Tuple{}) = error("")
```
Comment (GiggleLiu) [inline src/reorder.jl:33]: remove

**Review by GiggleLiu — COMMENTED** [review]
Good job, but maybe next time you should wait for review if it is not in urgent needed.
There are multiple lines not tested, and also some issues should be resolved.

**Pull request merged**